xconf-angular-admin/src/main/webapp/app/xconf/telemetry/telemetrytwoprofile/telemetrytwoprofile.import.controller.js

```diff
@@ -0,0 +1,136 @@
+        .controller('TelemetryTwoProfileImportController', controller);
+
+    controller.$inject=['$scope', '$log', '$uibModal', 'alertsService', 'utilsService', 'importService', 'telemetryTwoProfileService', 'paginationService'];
+
+    function controller($scope, $log, $modal, alertsService, utilsService, importService, telemetryTwoProfileService, paginationService) {
+        var vm = this;
+
+        vm.retrieveFile = retrieveFile;
+        vm.importProfile = importProfile;
+        vm.importAllProfiles = importAllProfiles;
+        vm.telemetryProfiles = null;
+        vm.wrappedProfiles = null;
+        vm.overwriteAll = overwriteAll;
+        vm.isOverwritten = false;
+        vm.viewTelemetryTwoProfile = viewTelemetryTwoProfile;
+        vm.paginationStorageKey = 'telemetryTwoProfilePageSize';
+        vm.pageSize = paginationService.getPageSize(vm.paginationStorageKey);
+        vm.pageNumber = paginationService.getPageNumber();
+        vm.selectPage = selectPage;
+        vm.getGeneralItemsNumber = getGeneralItemsNumber;
+        vm.progressBarControl = importService.progressBarControl;
+
+        $scope.$on('$locationChangeSuccess', function () {
+            if (paginationService.paginationSettingsInLocationHaveChanged(vm.pageNumber, vm.pageSize)) {
+                vm.pageSize = paginationService.getPageSize(vm.paginationStorageKey);
+                vm.pageNumber = paginationService.getPageNumber();
+                selectPage();
+            }
+        });
+
+        async function retrieveFile(fileName) {
+            vm.telemetryProfiles = null;
+            try {
+                let file = await importService.openFile(fileName, null, this);
+                vm.isOverwritten = false;
+                vm.wrappedProfiles = importService.prepareEntitiesFromFile(file);
+                selectPage();
+            } catch (e) {
+                alertsService.showError({message: e});
+            }
+        }
+
+        function importProfile(wrappedProfile) {
+            if (wrappedProfile.overwrite) {
+                telemetryTwoProfileService.updateTelemetryTwoProfile(wrappedProfile.entity)
+                    .then(function (resp) {
+                        handleSuccessfulUpdate(resp, wrappedProfile.entity);
+                        utilsService.removeSelectedItemFromListById(vm.wrappedProfiles, wrappedProfile.entity.id);
+                    }, function (error) {
+                        alertsService.showError({message: error.data.message, title: 'Exception'});
+                    });
+            } else {
+                telemetryTwoProfileService.createTelemetryTwoProfile(wrappedProfile.entity)
+                    .then(function () {
+                        alertsService.showSuccessMessage({message: wrappedProfile.entity.name + ' profile saved'});
+                        utilsService.removeSelectedItemFromListById(vm.wrappedProfiles, wrappedProfile.entity.id);
+                    }, function (error) {
```

Comment:
use a function reference here instead of full function declaration, like below

> alertsService.errorHandler

Reply:
fixed, thanks